File: solutions/cpp/Problem100.cpp
#pragma once
/*
100. Same Tree

Given two binary trees, write a function to check if they are the same or not.

Two binary trees are considered the same if they are structurally identical and the nodes have the same value.
*/
#include "Source.h"


class Solution {
public:
	bool isSameTree(TreeNode* p, TreeNode* q) {
		if (!p && !q) {
			return true;
		}
		else if ((!p && q) || (p && !q) || (p->val != q->val)) {
			return false;
		}
		else {
			return (isSameTree(p->left, q->left) && isSameTree(p->right, q->right));
		}
	}
	void test() {

	}
};

File: solutions/python3/problem100.py
# -*- coding: utf-8 -*-
"""
100. Same Tree

Given two binary trees, write a function to check if they are the same or not.

Two binary trees are considered the same if they are structurally identical and the nodes have the same value.
"""


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def isSameTree(self, p, q):
        if p is None and q is None:
            return True
        elif p is None and q is not None:
            return False
        elif q is None and p is None:
            return False
        else:
            if p.val != q.val:
                return False
            else:
                return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)

